nameserver/dns.go

```diff
@@ -286,9 +289,16 @@ func truncateResponse(response *dns.Msg, maxSize int) {
 	response.Truncated = true
 }
 
-func getMaxResponseSize(req *dns.Msg, defaultMaxResponseSize int) int {
+func (h *handler) responseTooBig(request, response *dns.Msg) bool {
+	if len(response.Answer) <= 1 || h.maxResponseSize <= 0 {
```

Comment:
This is the bug fix; checkout maxResponseSize > 0 before deciding if a response is too big.